webbpsf/webbpsf_core.py

```diff
@@ -300,9 +301,20 @@ def _getFITSHeader(self, result, options):
         result[0].header['DATAVERS'] =(self._data_version, "WebbPSF reference data files version")
 
         result[0].header['DET_NAME'] = (self.detector, "Name of detector on this instrument")
-        dpos = self.detector_position
-        result[0].header['DET_X'] = (dpos[0], "Detector X pixel position")
-        result[0].header['DET_Y'] = (dpos[1], "Detector Y pixel position")
+
+        # Correct detector pixel coordinates to allow for even arrays to be centered on half pixel boundary
+        dpos = copy.deepcopy(self.detector_position)
+        oversamp = result[0].header['OVERSAMP']
+        size = result[0].data.shape[0]
+
+        if size / oversamp % 2 == 0:  # even arrays must be at a half pixel
+            lst = list(dpos)
+            lst[0] += 0.5
+            lst[1] += 0.5
+            dpos = tuple(lst)
```

Comment:
I think this part can be simplified, by using a numpy array instead of list or tuple. 

```dpos = np.asarray(self.detector_position) +0.5```

Come to think of it, that probably eliminates the need for the deepcopy too, since the ndarray will be distinct from the tuple.

Reply:
Yes you're right, np.asarray would work better - I'll fix that. But you'd have to make the + 0.5 addition only under the if statement, while the dpos variable needs to be defined for both even and odd arrays (since that's the variable written into the header), so those parts do need to be separate